.github/workflows/Periodic-CI.yml

```diff
@@ -0,0 +1,237 @@
+name: Periodic-CI
+
+on:
+  # On demand execution of workflow will run all suites.
+  workflow_dispatch:
+    inputs:
+      test-targets:
+        required: true
+        description: 'Specify test markers to run'
+        default: '[\"pre_merge\", \"file_system_rules\", \"k8s_object_rules\", \"process_api_server_rules\", \"process_controller_manager_rules\", \"process_etcd_rules\", \"process_kubelet_rules\", \"process_scheduler_rules\"]'
+
+  schedule:
+    # Nightly job. Running every day at 2 am
+    - cron: '00 02 * * *'
+
+    # Weekly job running every saturday at 2 am
+    - cron: '00 02 * * 6'
+
+env:
+  K8S_MANIFEST_DIR: deploy/k8s/kustomize/base
+  GO_VERSION: 1.18.2
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  GITHUB_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  CONTAINER_SUFFIX: ${{ github.run_id }}
+
+jobs:
+  Setup:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    timeout-minutes: 15
+    steps:
+    - id: Nightly
+      if: github.event_name == 'schedule' && github.event.schedule == '00 02 * * *'
+      env:
+        TEST_TARGETS: '"[\"pre_merge\", \"file_system_rules\", \"k8s_object_rules\", \"process_api_server_rules\", \"process_controller_manager_rules\", \"process_etcd_rules\", \"process_kubelet_rules\", \"process_scheduler_rules\"]"'
+      run: echo "TEST_MATRIX=${TEST_TARGETS}" >> $GITHUB_ENV
+    - id: Weekly
+      if: github.event_name == 'schedule' && github.event.schedule != '00 02 * * 6'
+      env:
+        TEST_TARGETS: '[\"pre_merge\", \"file_system_rules\", \"k8s_object_rules\", \"process_api_server_rules\", \"process_controller_manager_rules\", \"process_etcd_rules\", \"process_kubelet_rules\", \"process_scheduler_rules\"]'
+      run: echo "TEST_MATRIX=${TEST_TARGETS}" >> $GITHUB_ENV
```

Comment:
According to the **if** statement weekly is defined by event 'schedule' and **not** '00 02 * * 6'.
What did you meant?

Reply:
These are two fields coming from the payload json.
github.event_name holds the event type. Push, pr etc. while github.event.schedule holds the cron data.

It allows us to identify a schedule trigger. Differentiate  between two different schedules to determine which tests should run on weekly vs nightly for example.
the `!=` is fixed